## **Inspection des 5 sources**

### **1. Importations des bibilothèques**

In [2]:
import pandas as pd
from pathlib import Path

### **2. Parcours des fichiers** 

In [3]:
DATA_DIR = Path("/mnt/c/Users/KSOMS/Favorites/awale_boissons/data/raw") #On va éviter d'écrire les chemins en dur partout dans le notebook

#### **2.1 campaign_spend_export**

In [4]:
campaign = DATA_DIR / "awale_boissons_starter_dataset.xlsx"
df_campaigns = pd.read_excel(campaign, sheet_name="campaign_spend_export")

In [5]:
df_campaigns.head()

,platform,campaign_name,date_start,date_end,spend,impressions,clicks,objective
0,Google Ads,awale can 2026 stories,2026-01-03,2026-01-30,98 200 FCFA,32961.0,1996.0,Engagement
1,TikTok,tiktok gingembre promo juin,04/06/2026,29/06/2026,28 100,37428.0,281.0,awareness
2,Meta,awale mai bouye trafic,2026-05-01,2026-05-30,248500,178467.0,2146.0,Trafic
3,TikTok,awale_gingembre_mars26,04/03/2026,29/03/2026,358 900,556571.0,4650.0,engagement
4,TikTok,tiktok bissap promo avril,05/04/2026,29/04/2026,226 000,316056.0,2329.0,awareness


In [6]:
df_campaigns.info()

<class 'pandas.DataFrame'>
RangeIndex: 72 entries, 0 to 71
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   platform       72 non-null     str    
 1   campaign_name  72 non-null     str    
 2   date_start     72 non-null     str    
 3   date_end       72 non-null     str    
 4   spend          72 non-null     str    
 5   impressions    66 non-null     float64
 6   clicks         66 non-null     float64
 7   objective      72 non-null     str    
dtypes: float64(2), str(6)
memory usage: 9.3 KB


- date_start et date_end sont des chaines de caractères et doivent être des variables de type date
- spend est au format chaine de caractère 

In [7]:
#Vérification des plateformes
df_campaigns["platform"].unique()

<ArrowStringArray>
['Google Ads', 'TikTok', 'Meta', 'Radio', 'Influenceur']
Length: 5, dtype: str

In [8]:
df_campaigns["platform"].value_counts()

platform
TikTok         31
Meta           24
Google Ads     11
Radio           3
Influenceur     3
Name: count, dtype: int64

- 31 campagnes publiciataires menées sur TikTok
- 24 campagnes publiciataires menées sur Meta
- 11 campagnes publiciataires menées sur Google Ads
- 3 campagnes publiciataires menées à la Radio
- 3 campagnes publiciataires menées par les Influenceurs

In [9]:
#Nombres de lignes et de colonnes
print(f"Nombre de lignes : {df_campaigns.shape[0]}")
print(f"Nombre de colonnes : {df_campaigns.shape[1]}")

Nombre de lignes : 72
Nombre de colonnes : 8


In [10]:
#Nom des colonnes
print("Nom des colonnes :")
for col in df_campaigns.columns:
    print(col)

Nom des colonnes :
platform
campaign_name
date_start
date_end
spend
impressions
clicks
objective


In [11]:
#Type des colonnes
print("Type des colonnes :")
for col in df_campaigns.columns:
    print(f"{col} : {df_campaigns[col].dtype}")

Type des colonnes :
platform : str
campaign_name : str
date_start : str
date_end : str
spend : str
impressions : float64
clicks : float64
objective : str


In [12]:
#Valeurs manquantes
print("Valeurs manquantes :")
df_campaigns.isnull().sum().sort_values(ascending=False)

Valeurs manquantes :


clicks           6
impressions      6
platform         0
campaign_name    0
date_end         0
date_start       0
spend            0
objective        0
dtype: int64

In [13]:
#Doublons
print("Doublons :")
df_campaigns.duplicated().sum()

Doublons :


np.int64(18)

Il y a 18 doublons

#### **2.2 pos_sales_daily**

In [14]:
pos_sales = DATA_DIR / "awale_boissons_starter_dataset.xlsx"
df_pos_sales = pd.read_excel(pos_sales, sheet_name="pos_sales_daily")

In [15]:
df_pos_sales.head()

,sale_date,pos_id,pos_name,commune,channel,product_sku,units_sold,revenue_fcfa
0,2026-01-01,POS001,Supermarché Angré 8e Tranche,Cocody,retail,BIS-1L,18,27000
1,2026-01-01,POS001,Supermarché Angré 8e Tranche,Cocody,retail,BIS-33,11,5500
2,2026-01-01,POS002,Boutique Riviera 3,Cocody,retail,BIS-33,10,5000
3,2026-01-01,POS004,Kiosque Deux Plateaux Vallon,Cocody,retail,BIS-1L,27,40500
4,2026-01-01,POS004,Kiosque Deux Plateaux Vallon,Cocody,retail,BIS-33,18,9000


In [56]:
len(df_pos_sales[df_pos_sales["revenue_fcfa"] < 0])

58

In [57]:
len(df_pos_sales[df_pos_sales["units_sold"] < 0])

58

In [16]:
df_pos_sales.info()

<class 'pandas.DataFrame'>
RangeIndex: 10631 entries, 0 to 10630
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   sale_date     10631 non-null  datetime64[us]
 1   pos_id        10631 non-null  str           
 2   pos_name      10631 non-null  str           
 3   commune       10631 non-null  str           
 4   channel       10631 non-null  str           
 5   product_sku   10631 non-null  str           
 6   units_sold    10631 non-null  int64         
 7   revenue_fcfa  10631 non-null  int64         
dtypes: datetime64[us](1), int64(2), str(5)
memory usage: 1.2 MB


In [17]:
#Vérification des communes
df_pos_sales["commune"].unique()

<ArrowStringArray>
[     'Cocody',    'Yopougon',     'Marcory', 'Treichville',      'COCODY',
    'YOPOUGON',     'MARCORY', 'TREICHVILLE']
Length: 8, dtype: str

Il y a des incohérence dans les nom enregistrés (majuscule et miniscule)

In [18]:
df_pos_sales["commune"].value_counts()

commune
Marcory        3082
Yopougon       2925
Cocody         2455
Treichville    1788
MARCORY         125
YOPOUGON        121
TREICHVILLE      69
COCODY           66
Name: count, dtype: int64

Après nettoyage on obtiendra le nombre réel

In [19]:
#Nombres de lignes et de colonnes
print(f"Nombre de lignes : {df_pos_sales.shape[0]}")
print(f"Nombre de colonnes : {df_pos_sales.shape[1]}")

Nombre de lignes : 10631
Nombre de colonnes : 8


In [20]:
#Nom des colonnes
print("Nom des colonnes :")
for col in df_pos_sales.columns:
    print(col)

Nom des colonnes :
sale_date
pos_id
pos_name
commune
channel
product_sku
units_sold
revenue_fcfa


In [21]:
#Type des colonnes
print("Type des colonnes :")
for col in df_pos_sales.columns:
    print(f"{col} : {df_pos_sales[col].dtype}")

Type des colonnes :
sale_date : datetime64[us]
pos_id : str
pos_name : str
commune : str
channel : str
product_sku : str
units_sold : int64
revenue_fcfa : int64


Chaque variable correspond a son type de données

In [22]:
#Valeurs manquantes
print("Valeurs manquantes :")
df_pos_sales.isnull().sum().sort_values(ascending=False)

Valeurs manquantes :


sale_date       0
pos_id          0
pos_name        0
commune         0
channel         0
product_sku     0
units_sold      0
revenue_fcfa    0
dtype: int64

In [23]:
#Doublons
print("Doublons :")
df_pos_sales.duplicated().sum()

Doublons :


np.int64(96)

Il y a 96 doublons

#### **2.3 social_comments**

In [24]:
social_comments = DATA_DIR / "awale_boissons_starter_dataset.xlsx"
df_social_comments = pd.read_excel(social_comments, sheet_name="social_comments")

In [25]:
df_social_comments.head()

,comment_id,platform,post_id,published_at,author_handle,comment_text,like_count,reply_to_id
0,C000014,Facebook,FA_20260101_03,2026-01-01 07:00:00,@ibrahim842,c'est gbé ça 😎 à Port-Bouët,2,NaN
1,C000016,Instagram,IN_20260101_01,2026-01-01 08:02:00,@traore321,Bjr les diabétiques peuvent boire ? à Abobo,3,NaN
2,C000012,Instagram,IN_20260101_01,2026-01-01 08:54:00,@didi913,love the branding 🔥,0,NaN
3,C000007,Instagram,IN_20260101_01,2026-01-01 08:55:00,@didi700,la pub pendant le match était bien placée à Po...,11,NaN
4,C000013,Facebook,FA_20260101_01,2026-01-01 12:49:00,@bakary406,"plus de bissap dans mon quartier, c'est comment ?",1,NaN


In [26]:
df_social_comments.info()

<class 'pandas.DataFrame'>
RangeIndex: 2856 entries, 0 to 2855
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   comment_id     2856 non-null   str           
 1   platform       2856 non-null   str           
 2   post_id        2856 non-null   str           
 3   published_at   2856 non-null   datetime64[us]
 4   author_handle  2856 non-null   str           
 5   comment_text   2856 non-null   str           
 6   like_count     2856 non-null   int64         
 7   reply_to_id    90 non-null     str           
dtypes: datetime64[us](1), int64(1), str(6)
memory usage: 395.1 KB


In [27]:
#Vérification des communes
df_social_comments["platform"].unique()

<ArrowStringArray>
['Facebook', 'Instagram', 'TikTok']
Length: 3, dtype: str

Les commentaires viennent de 3 plateforme :
- Facebook
- Instagram
- TikTok

In [28]:
df_social_comments["platform"].value_counts()

platform
Instagram    1138
Facebook      949
TikTok        769
Name: count, dtype: int64

On a :
- 1138 commentaires provenant d'Instagram
- 949 commentaires provenant de Facebook (Meta)
- 769 commentaires provenant de TikTok

In [29]:
#Nombres de lignes et de colonnes
print(f"Nombre de lignes : {df_social_comments.shape[0]}")
print(f"Nombre de colonnes : {df_social_comments.shape[1]}")

Nombre de lignes : 2856
Nombre de colonnes : 8


In [30]:
#Nom des colonnes
print("Nom des colonnes :")
for col in df_social_comments.columns:
    print(col)

Nom des colonnes :
comment_id
platform
post_id
published_at
author_handle
comment_text
like_count
reply_to_id


In [31]:
#Type des colonnes
print("Type des colonnes :")
for col in df_social_comments.columns:
    print(f"{col} : {df_social_comments[col].dtype}")

Type des colonnes :
comment_id : str
platform : str
post_id : str
published_at : datetime64[us]
author_handle : str
comment_text : str
like_count : int64
reply_to_id : str


In [32]:
#Valeurs manquantes
print("Valeurs manquantes :")
df_social_comments.isnull().sum().sort_values(ascending=False)

Valeurs manquantes :


reply_to_id      2766
comment_id          0
post_id             0
platform            0
published_at        0
author_handle       0
comment_text        0
like_count          0
dtype: int64

In [33]:
#Doublons
print("Doublons :")
df_social_comments.duplicated().sum()

Doublons :


np.int64(25)

Il y a 25 doublons

#### **2.4 whatsapp_orders**

In [34]:
whatsapp_orders = DATA_DIR / "awale_boissons_starter_dataset.xlsx"
df_whatsapp_orders = pd.read_excel(whatsapp_orders, sheet_name="whatsapp_orders")

In [35]:
df_whatsapp_orders.head()

,order_ref,received_at,customer_phone,items_text,amount_fcfa,delivery_zone,status
0,WA-2606-0034,01/06/2026,05 48 68 72 57,3 bouye 1L et 2 bissap 33cl,5600.0,marcory z4,Livré
1,WA-2601-1369,02/01/2026,+2250755364843,3 bissap 1L (fraîches svp),4500.0,Treichville Arras,annulé
2,WA-2601-1029,02/01/2026,0144564506,3 bissap 1L (fraîches svp),NaN,Yop Selmer,annulée
3,WA-2602-0192,02/02/2026,+2250749480972,2 bouye + 1 gingembre 1L,NaN,Koumassi,livrée
4,WA-2602-0629,02/02/2026,05 32 12 43 01,5 bissap 33cl,7200.0,Bingerville,LIVRE


In [36]:
df_whatsapp_orders.info()

<class 'pandas.DataFrame'>
RangeIndex: 1066 entries, 0 to 1065
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   order_ref       1066 non-null   str    
 1   received_at     1066 non-null   str    
 2   customer_phone  1066 non-null   str    
 3   items_text      1066 non-null   str    
 4   amount_fcfa     935 non-null    float64
 5   delivery_zone   1066 non-null   str    
 6   status          1066 non-null   str    
dtypes: float64(1), str(6)
memory usage: 141.1 KB


In [58]:
df_whatsapp_orders[df_whatsapp_orders["order_ref"]=="WA-2601-0262"]

,order_ref,received_at,customer_phone,items_text,amount_fcfa,delivery_zone,status
84,WA-2601-0262,2026-01-01 18:52,01 86 23 10 63,commande habituelle x1,1600.0,riviera3,annulé
185,WA-2601-0262,2026-01-24 14:05,01 15 45 92 51,commande habituelle x5,8200.0,angre 8eme tranche,livrée


In [37]:
#Vérification des zones de livraison
df_whatsapp_orders["delivery_zone"].unique()

<ArrowStringArray>
[        'marcory z4',  'Treichville Arras',         'Yop Selmer',
           'Koumassi',        'Bingerville',      'Cocody Centre',
    'yopougon selmer',      'Deux Plateaux',          'RIVIERA 3',
            'Plateau',  '2 plateaux vallon',             'Adjamé',
           'Angré 8e',     'Marcory Zone 4',           'riviera3',
          'Riviera 3', 'angre 8eme tranche',        'II Plateaux']
Length: 18, dtype: str

In [38]:
df_whatsapp_orders["delivery_zone"].value_counts()

delivery_zone
Adjamé                70
angre 8eme tranche    65
Koumassi              64
Deux Plateaux         64
Angré 8e              63
Yop Selmer            62
marcory z4            61
Cocody Centre         61
II Plateaux           61
RIVIERA 3             60
Riviera 3             60
2 plateaux vallon     59
Marcory Zone 4        58
Treichville Arras     54
Bingerville           54
yopougon selmer       54
Plateau               49
riviera3              47
Name: count, dtype: int64

In [39]:
#Nombres de lignes et de colonnes
print(f"Nombre de lignes : {df_whatsapp_orders.shape[0]}")
print(f"Nombre de colonnes : {df_whatsapp_orders.shape[1]}")

Nombre de lignes : 1066
Nombre de colonnes : 7


Il y a eu 1066 commandes sur WhatsApp

In [40]:
#Nom des colonnes
print("Nom des colonnes :")
for col in df_whatsapp_orders.columns:
    print(col)

Nom des colonnes :
order_ref
received_at
customer_phone
items_text
amount_fcfa
delivery_zone
status


In [41]:
#Type des colonnes
print("Type des colonnes :")
for col in df_whatsapp_orders.columns:
    print(f"{col} : {df_whatsapp_orders[col].dtype}")

Type des colonnes :
order_ref : str
received_at : str
customer_phone : str
items_text : str
amount_fcfa : float64
delivery_zone : str
status : str


In [42]:
#Valeurs manquantes
print("Valeurs manquantes :")
df_whatsapp_orders.isnull().sum().sort_values(ascending=False)

Valeurs manquantes :


amount_fcfa       131
received_at         0
order_ref           0
customer_phone      0
items_text          0
delivery_zone       0
status              0
dtype: int64

amount_fcfa possède 131 valeurs manquantes

In [43]:
#Doublons
print("Doublons :")
df_whatsapp_orders.duplicated().sum()

Doublons :


np.int64(0)

Il y a 0 doublons

#### **2.5 media_plan**

In [44]:
media_plan = DATA_DIR / "awale_boissons_starter_dataset.xlsx"
df_media_plan = pd.read_excel(media_plan, sheet_name="media_plan")

In [45]:
df_media_plan.head()

,plan_id,month,channel,planned_budget_fcfa,invoiced_fcfa,objective,owner,notes
0,MP-001,Janvier 2026,FB/IG,1150000,1212000.0,Conversions,K. Bamba,NaN
1,MP-002,Janvier 2026,TikTok,600000,640000.0,Notoriété / trafic,K. Bamba,NaN
2,MP-003,Janvier 2026,Google,300000,298500.0,Marque,S. Diallo,NaN
3,MP-004,Janvier 2026,Radio,900000,950000.0,Notoriété,A. Kouassi,"spot 30s Trace FM + Nostalgie, forte rotation ..."
4,MP-005,Janvier 2026,Influenceurs,400000,400000.0,Notoriété,K. Bamba,NaN


In [46]:
df_media_plan.info()

<class 'pandas.DataFrame'>
RangeIndex: 36 entries, 0 to 35
Data columns (total 8 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   plan_id              36 non-null     str    
 1   month                36 non-null     str    
 2   channel              36 non-null     str    
 3   planned_budget_fcfa  36 non-null     int64  
 4   invoiced_fcfa        26 non-null     float64
 5   objective            36 non-null     str    
 6   owner                36 non-null     str    
 7   notes                13 non-null     str    
dtypes: float64(1), int64(1), str(6)
memory usage: 4.5 KB


In [47]:
#Vérification des zones de livraison
df_media_plan["channel"].unique()

<ArrowStringArray>
['FB/IG', 'TikTok', 'Google', 'Radio', 'Influenceurs', 'Activation terrain']
Length: 6, dtype: str

In [48]:
df_media_plan["channel"].value_counts()

channel
FB/IG                 6
TikTok                6
Google                6
Radio                 6
Influenceurs          6
Activation terrain    6
Name: count, dtype: int64

In [49]:
#Nombres de lignes et de colonnes
print(f"Nombre de lignes : {df_media_plan.shape[0]}")
print(f"Nombre de colonnes : {df_media_plan.shape[1]}")

Nombre de lignes : 36
Nombre de colonnes : 8


Il y a eu 36 plan média pour le client

In [50]:
#Nom des colonnes
print("Nom des colonnes :")
for col in df_media_plan.columns:
    print(col)

Nom des colonnes :
plan_id
month
channel
planned_budget_fcfa
invoiced_fcfa
objective
owner
notes


In [51]:
#Type des colonnes
print("Type des colonnes :")
for col in df_media_plan.columns:
    print(f"{col} : {df_media_plan[col].dtype}")

Type des colonnes :
plan_id : str
month : str
channel : str
planned_budget_fcfa : int64
invoiced_fcfa : float64
objective : str
owner : str
notes : str


In [52]:
#Valeurs manquantes
print("Valeurs manquantes :")
df_media_plan.isnull().sum().sort_values(ascending=False)

Valeurs manquantes :


notes                  23
invoiced_fcfa          10
plan_id                 0
month                   0
planned_budget_fcfa     0
channel                 0
objective               0
owner                   0
dtype: int64

- 23 valeurs manquantes pour notes
- 10 valeurs manquantes pour invoiced_fcfa          

In [53]:
#Doublons
print("Doublons :")
df_media_plan.duplicated().sum()

Doublons :


np.int64(0)

Il y a 0 doublons

## **3. Récapitulatif**

| Source | Nombres de lignes | Nombres de colonnes | Nombres de valeur manquante | Nombres de doublons | Anomalies |
|:-------|:--------:| --------:|:--------:|:--------:|:--------:|
| campaign_spend_export  | 72 | 8 | 12 | 18  | Absence de campaign_id, noms incohérents, dates mixtes, spend en FCFA/EUR, impressions et clics absents pour la radio et les influenceurs, et deux mois dupliqués |
| post_sales_daily  | 10631 | 8 | 0 | 96 | Deux semaines consécutives manquantes à cause d'une panne, noms de magasins variables, pos_id instable et revenus négatifs correspondant aux retours |
| social_comments  | 2856 | 8 | 2766 | 25  | Français, Nouchi et anglais mélangés, emojis et abréviations, environ 15 % de spam ou de contenu non pertinent, et absence de liaison directe entre un post et une campagne |
| whatsapp_orders  | 1066 | 7 | 131 | 0 | Téléphones sous plusieurs formats, items en prose, montant manquant sur environ 1 commande sur 8, et order_ref répété |
| median_plan  | 36 | 8 | 33 | 0 | FB/IG comptabilisés séparément de Meta, radio regroupée alors que les stations sont facturées séparément, dépenses radio/influenceurs/activation présentes uniquement ici, et absence de réconciliation avec campaign_spend_export  |